test/distributed/org/apache/cassandra/distributed/test/ReadRepairTest.java

```diff
@@ -127,61 +211,197 @@ public void movingTokenReadRepairTest() throws Throwable
+                                              "{'class': 'SimpleStrategy', 'replication_factor': 1}"));
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.t (k int PRIMARY KEY, a int, b int)" +
+                                              " WITH read_repair='blocking'"));
+
+            // insert a row that will only get to one node due to the RF=1
+            Object[] row = row(1, 1, 1);
+            cluster.get(1).executeInternal(withKeyspace("INSERT INTO %s.t (k, a, b) VALUES (?, ?, ?)"), row);
+
+            // flush to ensure reads come from sstables
+            cluster.get(1).flush(KEYSPACE);
+
+            // at RF=1 it shouldn't matter which node we query, as the data should always come from the only replica
+            String query = withKeyspace("SELECT * FROM %s.t WHERE k = 1");
+            for (int i = 1; i <= cluster.size(); i++)
+                assertRows(cluster.coordinator(i).execute(query, ALL), row);
+
+            // at RF=1 the prevoius queries shouldn't have triggered read repair
+            assertRows(cluster.get(1).executeInternal(query), row);
+            assertRows(cluster.get(2).executeInternal(query));
+            assertRows(cluster.get(3).executeInternal(query));
+
+            // alter RF
+            System.setProperty(Config.PROPERTY_PREFIX + "allow_alter_rf_during_range_movement", "true");
+            cluster.schemaChange(withKeyspace("ALTER KEYSPACE %s WITH replication = " +
+                                              "{'class': 'SimpleStrategy', 'replication_factor': 3}"));
+
+            // altering the RF shouldn't have triggered any read repair
+            assertRows(cluster.get(1).executeInternal(query), row);
+            assertRows(cluster.get(2).executeInternal(query));
+            assertRows(cluster.get(3).executeInternal(query));
+
+            // query just column 'a' with CL=ALL to trigger partial read repair on that column
+            assertRows(cluster.coordinator(2).execute(withKeyspace("SELECT a FROM %s.t WHERE k = 1"), ALL), row(1));
+
+            // check each replica individually to verify that read repair has partially repaired the replicas
+            Object[] partialRow = row(1, 1, null);
+            assertRows(cluster.get(1).executeInternal(query), row);
+            assertRows(cluster.get(2).executeInternal(query), partialRow);
+            assertRows(cluster.get(3).executeInternal(query), partialRow);
+
+            // query again at CL=ALL but this time selecting all columns, which should ensure that 'b' gets repaired
+            assertRows(cluster.coordinator(3).execute(query, ALL), row);
+
+            // check each replica individually again now that we expect the data to be fully repaired
+            assertRows(cluster.get(1).executeInternal(query), row);
+            assertRows(cluster.get(2).executeInternal(query), row);
+            assertRows(cluster.get(3).executeInternal(query), row);
```

Comment:
It seems like there are a lot of things going on in this test. I suppose there might be some concerns about cluster reuse, but I'm wondering if things like making sure read repair on a single column only partially repairs a row could be the subject of their own test.

I'm also not entirely clear about why changing the RF was an essential part of verifying the behavior from CASSANDRA-10655 and CASSANDRA-10657. Making sure altering the RF doesn't trigger RR seems like it could be it's own test, but moving from RF=1 to RF=3 seems more or less like starting at RF=3 and doing an internal write to only one node.

Reply:
@maedhroz I think that the original purpose of the ticket was only testing partial row RR, and altering the RF was only a means to simulate the scenario. Partial RR should be widely covered by `ReadRepairQueryTypesTest`, but I still see some value in testing basic RR behaviour going from RF=1 to RF>1. I allows us to verify that:
* With RF=1 reads doesn't trigger any sort of repair.
* Altering the RF doesn't repair anything.
* After increasing the RF read repair starts to work as usual.
I don't think that those things are checked elsewhere. I have simplified the test removing the partial row read and updated the doc. WDYT?